### PR TITLE
Rest package: reduce test runtime

### PR DESCRIPTION
**Pull request by xpivarc**
### What this PR does
Removes unnecessary `time.Sleep`. This reduce the runtime from ~40 seconds down to ~2 seconds

Fixes #

Links to places where the discussion took place: 

### Release note

```release-note
NONE
```

### CODE REVIEW

**Review by fossedihelm — COMMENTED** [review]
Looks good!
Just a small improvement

File: pkg/virt-api/rest/streamer_test.go [hunk 1]
```diff
@@ -105,6 +107,13 @@ var _ = Describe("Streamer", func() {
 		params[definitions.NamespaceParamName] = testNamespace
 		params[definitions.NameParamName] = testName
 
+		streamer.dialer.fetchVMI = func(namespace, name string) (*v1.VirtualMachineInstance, *errors.StatusError) {
+			Expect(namespace).To(Equal(testNamespace))
```
[the remark this answers is not in the document]
Comment (jschintag) [reply pkg/virt-api/rest/streamer_test.go:111]: Yes, there should be a `defer GinkgoRecover()` here.

File: pkg/virt-api/rest/streamer_test.go [hunk 2]
```diff
@@ -219,14 +228,13 @@ var _ = Describe("Streamer", func() {
...
-			call <- struct{}{}
-		}
+	It("returns if the client connection upgrade failed", func() {
 		Expect(streamer.Handle(req, resp)).To(HaveOccurred())
-		Consistently(call, defaultTestTimeout).ShouldNot(Receive())
+		response, err := io.ReadAll(respRecorder.Body)
+		Expect(err).To(Not(HaveOccurred()))
+		Expect(string(response)).To(ContainSubstring("the client is not using the websocket protocol"))
 	})
+
```
Comment (jschintag) [inline pkg/virt-api/rest/streamer_test.go:237]: Nit: There is no space between the other tests, should keep it equal

File: pkg/virt-api/rest/streamer_test.go [hunk 3]
```diff
@@ -308,12 +316,13 @@ var _ = Describe("Streamer", func() {
 	It("closes clientSocket when keepAliveClient cancels context", func() {
 		streamer.streamToClient = func(clientSocket *websocket.Conn, serverConn net.Conn, result chan<- streamFuncResult) {
 			streamToClientCalled <- struct{}{}
-			time.Sleep(defaultTestTimeout * 2)
+			_, _ = io.Copy(io.Discard, serverConn)
+
```
Comment (jschintag) [inline pkg/virt-api/rest/streamer_test.go:320]: Nit: Same here, seems an extra blank line slipped in.

**Review by jschintag — COMMENTED** [review]
Same for me, it looks good, just small nit's

File: pkg/virt-api/rest/streamer_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (xpivarc) [reply pkg/virt-api/rest/streamer_test.go:111]: I made the test fail and this is the output
```
 [FAILED] Expected
      <string>: test-namespace
  not to equal
      <string>: test-namespace
  In [It] at: /home/le_pivo/Desktop/kubevirt/pkg/virt-api/rest/streamer_test.go:111 @ 08/01/24 12:28:21
```
"GinkgoRecover should be deferred at the top of any spawned goroutine that (may) call `Fail`"
as this is not run in goroutine we are safe.

**Comment by xpivarc** [comment]
@jschintag @fossedihelm PTAL

**Comment by jschintag** [comment]
/lgtm

Thank you @xpivarc

**Review by fossedihelm — APPROVED** [review]

**Pull request merged**